src/applications/disability-benefits/686c-674/config/chapters/674/student-last-term-information/studentLastTerm.js

```diff
@@ -0,0 +1,180 @@
+import currentOrPastDateUI from 'platform/forms-system/src/js/definitions/currentOrPastDate';
+import cloneDeep from 'platform/utilities/data/cloneDeep';
+import { TASK_KEYS } from '../../../constants';
+import { genericSchemas } from '../../../generic-schema';
+import { isChapterFieldRequired } from '../../../helpers';
+import { addressSchema } from '../../../address-schema';
+
+import { StudentNameHeader } from '../helpers';
+
+const { date, genericTextInput } = genericSchemas;
+
+const nonMilitaryAddressSchema = obj => {
+  const schema = cloneDeep(obj);
+  delete schema.properties['view:livesOnMilitaryBase'];
+  delete schema.properties['view:livesOnMilitaryBaseInfo'];
+  return schema;
+};
+
+const addressSchemaObj = nonMilitaryAddressSchema(addressSchema);
+
+export const schema = {
+  type: 'object',
+  properties: {
+    studentDidAttendSchoolLastTerm: {
+      type: 'boolean',
+    },
+    lastTermSchoolName: genericTextInput,
+    lastTermSchoolAddress: addressSchemaObj,
+    dateTermBegan: date,
+    dateTermEnded: date,
+    lastTermClassesPerWeek: {
+      type: 'number',
+    },
+    lastTermHoursPerWeek: {
+      type: 'number',
+    },
+  },
+};
+
+export const uiSchema = {
+  'ui:title': StudentNameHeader,
+  studentDidAttendSchoolLastTerm: {
+    'ui:title': 'Did student attend school last term?',
+    'ui:widget': 'yesNo',
+    'ui:required': formData =>
+      isChapterFieldRequired(formData, TASK_KEYS.report674),
+  },
+  lastTermSchoolName: {
+    'ui:required': formData => formData.studentDidAttendSchoolLastTerm === true,
+    'ui:title': 'Last term school’s name',
+    'ui:options': {
+      expandUnder: 'studentDidAttendSchoolLastTerm',
+      expandUnderCondition: true,
+    },
+  },
+  lastTermSchoolAddress: {
+    'ui:title': 'Last term school’s address',
+    'ui:options': {
+      expandUnder: 'studentDidAttendSchoolLastTerm',
+      expandUnderCondition: true,
+    },
+    countryName: {
+      'ui:required': formData =>
+        formData.studentDidAttendSchoolLastTerm === true,
+      'ui:title': 'Country',
+    },
+    addressLine1: {
+      'ui:required': formData =>
+        formData.studentDidAttendSchoolLastTerm === true,
+      'ui:title': 'Street',
+      'ui:errorMessages': {
+        required: 'Street address is required',
+        pattern: 'Street address must be under 100 characters',
+      },
+    },
+    addressLine2: {
+      'ui:title': 'Line 2',
+    },
+    addressLine3: {
+      'ui:title': 'Line 3',
+    },
+    city: {
+      'ui:required': formData =>
+        formData.studentDidAttendSchoolLastTerm === true,
+      'ui:errorMessages': {
+        required: 'City is required',
+        pattern: 'City must be under 100 characters',
+      },
+    },
+    stateCode: {
+      'ui:required': formData =>
+        formData?.lastTermSchoolAddress?.countryName === 'United States' &&
```

Comment:
I ran into an issue where fields contingent on the presence of a value in another address field threw an 'undefined` type error, but only in an address that expanded under another field based on a different formData condition. Optional chaining mitigates this issue.